design/protocol-changes/object-diffusion.tex

```diff
@@ -82,8 +82,10 @@ \subsection{Parameters}
 \begin{description}
 \item [\argfont{object}] The abstract type of objects diffused by the protocol.
 \item [\argfont{id}] Identifier that uniquely identifies an object.
-\item [\argfont{objectMetadata}] Additional information that is returned by the
-  server when querying for new object ids. For instance, the size of the object.
+\item [\argfont{objectIds}] An opaque type returned by the server when asked for object ids.
+  It is not explicitely a list of $\argfont{id}$, because the server may want to add metadata to the ids,
+  and/or run a compression scheme to limit the size of its response.
+\item [\argfont{responseToIds}] A function with type $\argfont{obejctIds} \rightarrow [\argfont{id}]$.
```

Comment:
To be generic and have a chance to cover Tx-Submission, shouldn't it then be
```
objectIds -> [(id, size)]
```
?

Also typo: s/obejct/object/

Reply:
I think we'll see in the implementation phase that the client needs to be careful about the size of its requests, and therefore it must know the size of the objects. For votes/certs, the size might be constant, so not a problem. For transactions, it isn't and it isn't derivable from the ids alone. However, I see your point; let's cross this bridge when we come to it and just merge this in the meantime.